pyproject.toml

```diff
@@ -167,6 +167,27 @@ extend_skip_glob = ["asdf/extern/*", ".eggs/*", ".tox/*"]
 
 [tool.ruff]
 line-length = 120
+select = ["ALL"]
+extend-ignore = [
+    # Ignored check groups
+    "C90", # mccabe complexity
+    "D", # pydocstyle
```

Comment:
We should slowly address all the errors this points out too. The majority of the errors were pointing out missing documentation on "public" methods/functions. We should either document these things or make them private in ASDF 3.0

Reply:
Note there are 2155 errors found by this one.